pkg/webhook/add_label_on_review_approval.go

```diff
@@ -17,42 +17,35 @@ package webhook
 import (
 	"context"
 	"fmt"
-	"net/http"
 	"strings"
 
 	"github.com/google/go-github/github"
 	"github.com/pkg/errors"
 	"github.com/syndesisio/pure-bot/pkg/config"
+	"go.uber.org/zap"
 )
 
-var addLabelOnReviewApprovalHandler Handler = &addLabelOnReviewApproval{}
-
 const (
-	approvedLabel       = "approved"
-	approvedReviewState = "approved"
+	approvedLabel  = "approved"
+	assignedAction = "approved"
 )
 
 type addLabelOnReviewApproval struct{}
 
-func (h *addLabelOnReviewApproval) HandleEvent(w http.ResponseWriter, eventObject interface{}, ghClientFunc GitHubAppsClientFunc, config config.GitHubAppConfig) error {
+func (h *addLabelOnReviewApproval) EventTypesHandled() []string {
+	return []string{"pull_request_review"}
+}
+
+func (h *addLabelOnReviewApproval) HandleEvent(eventObject interface{}, gh *github.Client, config config.GitHubAppConfig, logger *zap.Logger) error {
 	event, ok := eventObject.(*github.PullRequestReviewEvent)
 	if !ok {
 		return errors.New("wrong event eventObject type")
 	}
 
-	if event.Installation == nil {
+	if strings.ToLower(event.Review.GetState()) != assignedAction {
```

Comment:
Are we checking if the PR has been approved or assigned to someone?

Reply:
It's all about approval